src/components/engine/create-engine-instance.tsx

```diff
@@ -105,24 +115,22 @@ export const CreateEngineInstanceButton = ({
 
       {disclosure.isOpen && (
         <CreateCloudHostedEngineModal
-          hasValidPayment={
-            accountQuery.data?.status === AccountStatus.ValidPayment
-          }
           disclosure={disclosure}
-          onConfirm={async () => {
-            await requestCloudHostedEngine();
+          onClickAddToPlan={async (tier: EngineTier) => {
+            setEngineTier(tier);
             disclosure.onClose();
-          }}
-          onAddPaymentMethod={() => {
-            trackEvent({
-              category: "engine",
-              action: "click",
-              label: "clicked-add-payment-method",
-            });
-
-            // Switch the Cloud-hosted Engine modal with the Add Payment modal.
-            disclosure.onClose();
-            paymentDisclosure.onOpen();
+
+            if (accountQuery.data?.status === AccountStatus.ValidPayment) {
+              await addCloudHostedEngine();
```

Comment:
when this gets called the `setEngineTier()` from above is not set yet meaning the `engineTier` inside of that function will potentially use an incorrect value

Reply:
for a hacky fix pass the tier here directly
```suggestion
              await addCloudHostedEngine(tier);
```